### PR TITLE
CLOUDP-70032: mongocli atlas integration(s) create|update NEW_RELIC

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-70032

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli atlas integrations create NEW_RELIC --accountId 2857403 --licenceKey XXXXXXXXXXXX --readToken NRIQ-XXXXXXXXX --writeToken NRII-zpwCnhrDyUojdSJOk2LXXXXXXXXX -o json
{
  "links": [
    {
      "rel": "self",
      "href": "https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/5e4e593f70dfbf1010295836/integrations/NEW_RELIC?pageNum=1\u0026itemsPerPage=100"
    }
  ],
  "results": [
    {
      "type": "NEW_RELIC",
      "licenseKey": "eu01xx5f87268ff5028e69a6XXXXXXXXX",
      "accountId": "2857403",
      "writeToken": "NRII-zpwCnhrDyUojdSJOk2LDvtTXXXXXXX",
      "readToken": "NRIQ-MHCu7bPq2xvrrpVvybXXXXXXXX"
    }
  ],
  "totalCount": 1
}

```

```
./bin/mongocli atlas integrations create NEW_RELIC --accountId 2857403 --licenceKey eu01xx5f87268ff5028e69aXXXXXX --readToken NRIQ-MHCu7bPq2xvrrpVvybtXXXXXXX --writeToken NRII-zpwCnhrDyUojdSJOk2LDXXXXXXXXXX
New Relic integration created.
```

Screenshot 2020-09-03 at 13 02 38

### CODE REVIEW

File: internal/cli/atlas/integrations/create/create.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+package create
+
+import (
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/spf13/cobra"
+)
+
+func Builder() *cobra.Command {
+	const use = "create"
+	aliases := append(cli.GenerateAliases(use), cli.GenerateAliases("update")...)
```
Comment (gssbzn) [inline internal/cli/atlas/integrations/create/create.go:24]: [q] why `GenerateAliases`? it only works for plurals and camelcase so the aliases of create and update are just create and update

File: internal/cli/atlas/integrations/create/description.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package create
+
+const (
+	short    = "Create operations"
+	newRelic = "Create|Update New Relic integration"
```
Comment (gssbzn) [inline internal/cli/atlas/integrations/create/description.go:18]: please be human friendly, 
```suggestion
	CreateNewRelic = "Create or update the New Relic integration"
```

File: internal/cli/atlas/integrations/create/new_relic.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		ReadToken:  opts.readToken,
+	}
+}
+
+// mongocli atlas integration(s) create NEW_RELIC --licenceKey licenceKey --accountId accountId --writeToken writeToken --readToken readToken [--projectId projectId]
+func NewRelicBuilder() *cobra.Command {
+	opts := &NewRelicOpts{}
+	cmd := &cobra.Command{
+		Use:     "NEW_RELIC",
+		Aliases: []string{"new_relic", "newRelic", "nr", "NR"},
```
Comment (gssbzn) [inline internal/cli/atlas/integrations/create/new_relic.go:70]: I'd rather not use `"nr", "NR"` but I don't feel strong about it (it's two short for this type of command)

File: internal/cli/atlas/integrations/description.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package integrations
+
+const (
+	short = "Integration operations"
```
Comment (gssbzn) [inline internal/cli/atlas/integrations/description.go:17]:
```suggestion
	short = "Third party integration operations"
```

**Review by gssbzn — CHANGES_REQUESTED** [review]
just a couple of small nits

File: internal/cli/atlas/integrations/create/description.go (hunk 2 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/integrations/create/description.go:18]: Create is also the name of the package 😿 
Screenshot 2020-09-03 at 16 48 05

Comment (gssbzn) [reply internal/cli/atlas/integrations/create/description.go:18]: oh yeah sorry, I think I copy/pasted an old version you had... I don't mind the name just mond the content, your old name is fine

File: internal/cli/atlas/integrations/create/new_relic.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+func (opts *NewRelicOpts) Run() error {
+	r, err := opts.store.CreateIntegration(opts.ConfigProjectID(), newRelicIntegrationType, opts.newThirdPartyIntegration())
+	if err != nil {
+		return err
+	}
+	return opts.Print(r)
+}
+
+func (opts *NewRelicOpts) newThirdPartyIntegration() *atlas.ThirdPartyIntegration {
```
Comment (gssbzn) [inline internal/cli/atlas/integrations/create/new_relic.go:55]: [non-blocking] I would call this `newNewRelicIntegration` but we can do that later

File: internal/cli/atlas/integrations/create/create.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"github.com/spf13/cobra"
+)
+
+func Builder() *cobra.Command {
+	const use = "create"
+	cmd := &cobra.Command{
+		Use:     use,
+		Aliases: []string{"update"},
```
Comment (gssbzn) [inline internal/cli/atlas/integrations/create/create.go:25]: what happens if you call create multiple times? does it error?
Originally I defined this as create|update as the options are the same but you may need a different store for update

File: internal/cli/atlas/integrations/create/new_relic.go [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+	store      store.IntegrationCreator
+}
+
+func (opts *NewRelicOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+var createTemplate = "New Relic integration created.\n"
```
Comment (gssbzn) [inline internal/cli/atlas/integrations/create/new_relic.go:45]: depending on my previous question, if you can update using the create store then we should reword this, maybe something like `"New Relic integration configured.\n" or something

**Review by gssbzn — COMMENTED** [review]
Left a question, update pending on answer

File: internal/cli/atlas/integrations/create/create.go (hunk 6 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/integrations/create/create.go:25]: I am using this endpoint https://docs.atlas.mongodb.com/reference/api/third-party-integration-settings-update/. If the user runs this command multiple times the service details will be updated all the time

Screenshot 2020-09-03 at 17 42 22

Comment (gssbzn) [reply internal/cli/atlas/integrations/create/create.go:25]: and if doesn't exists will it create it?

Comment (andreaangiolillo) [reply internal/cli/atlas/integrations/create/create.go:25]: yes it does

File: internal/cli/atlas/integrations/create/new_relic.go (hunk 7 above)
Comment (gssbzn) [reply internal/cli/atlas/integrations/create/new_relic.go:45]: @andreaangiolillo may want to take a look at this then

**Comment by andreaangiolillo** [comment]
@gssbzn ready for review

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**